examples/temporal_graph_learning/model_utils.py

```diff
@@ -0,0 +1,411 @@
+
+def rel_name_map(rel_name):
+    """
+    Remove the number suffix from relation name.
+
+        {rel_name}_{i}-rev --> {rel_name}-rev
+        {rel_name}_{i} --> {rel_name}
+
+    Parameters
+    ----------
+    rel_name: string
+        relation name with number suffix
+
+    Returns
+    -------
+        string: relation name without number suffix
+    """
+    if "-rev" in rel_name:
+        return (
+            rel_name.split("-")[0].rstrip("_0123456789") + "-rev"
+        )
+    return rel_name.rstrip("_0123456789")
+
+
+def get_unique_rel_names(rel_names):
+    """
+    Given a list of relation name, remove the number suffix from each relation name.
+    This function calls `rel_name_map` on each relation name in the list.
+
+    Parameters
+    ----------
+    rel_names: list of string
+        list of relation name with number suffix
+
+    Returns
+    -------
+        list of string: list of relation name without number suffix
+    """
+    rel_names = [rel_name_map(rel_name) for rel_name in rel_names]
+    rel_names = list(set(rel_names))
+    rel_names.sort()
+    return rel_names
+
+def get_unique_etype_triplet(etype_triplet):
+    """
+    Given a list of etype triplet (src, relation_name, dst), remove the number suffix from each relation name.
+    This function calls `rel_name_map` on the relation name of each etype triplet in the list.
+
+    Parameters
+    ----------
+    etype_triplet: list of triplet
+        list of etype triplet (src, relation_name, dst) with number suffix
+
+    Returns
+    -------
+        list of triplet: list of etype triplet (src, relation_name, dst) without number suffix
+    """
+    # remove the timestamp from relation name, then form as triplet (src, rel, dst)
+    etype_triplet = [(src, rel_name_map(rel), dst) for src, rel, dst in etype_triplet]
+    etype_triplet = list(set(etype_triplet))
+    etype_triplet.sort()
+    return etype_triplet
+
+def rel_field_map(rel_name):
+    """
+    Return the number suffix from relation name.
+
+        {rel_name}_{i}-rev --> i
+        {rel_name}_{i} --> i
+
+    Parameters
+    ----------
+    rel_name: string
+        relation name with number suffix
+
+    Returns
+    -------
+        int: number suffix of each relation name
+    """
+    t = re.findall(r"\d+", rel_name)[-1]
+    assert t.isnumeric()
+    return t
+
+
+def get_unique_nfields(rel_names):
+    """
+    Given a list of relation name, return the number suffix from each relation name.
+    This function calls `rel_field_map` on each relation name in the list.
+
+    Parameters
+    ----------
+    rel_names: list of string
+        list of relation name with number suffix
+
+    Returns
+    -------
+        list of int: list of sorted number suffix of each relation name
+    """
+    field_ids = [rel_field_map(rel_name) for rel_name in rel_names]
+    field_ids = list(set(field_ids))
+    field_ids.sort()
+    return field_ids
+
+def get_merge_canonical_etype_mapping(etypes):
```

Comment:
i think you should rename it. it doesn't suggest this is to get the mapping for temporal aggregation. to make the name shorter, you can remove canonical.

Reply:
I renamed it to `get_temporal_ordered_etypes`